lib/chibi/sxml.scm

```diff
@@ -80,13 +80,24 @@
   (call-with-output-string
     (lambda (out) (html-display-escaped-string str out))))
 
+(define indentable-elements
+  '(address article aside base blockquote body dd details dialog
+    div dl dt fieldset figcaption figure footer form h1 h2 h3 h4
+    h5 h6 head header hgroup hr li link main meta nav ol p pre
+    script section style table title ul))
+
+(define (indent i out)
+  (do ((j (* 2 i) (- j 1))) ((= j 0)) (write-char #\space out)))
+
 ;;> Render (valid, expanded) \var{sxml} as html.
 ;;> \var{@raw} tag is considered safe text and not processed or escaped.
 (define (sxml-display-as-html sxml . o)
-  (let ((out (if (pair? o) (car o) (current-output-port))))
+  (let-optionals o ((out (current-output-port))
+		    (indent? #false))
     (let lp ((sxml (if (and (pair? sxml) (eq? '*TOP* (car sxml)))
                        (cdr sxml)
-                       sxml)))
+                       sxml))
+	     (i 0))
```

Comment:
nit: i sounds like a normal counter, maybe better to name this depth or indent?

Reply:
Good idea.  I renamed it `depth`.